Test/HUnit/Lang.hs

```diff
@@ -89,4 +91,8 @@ performTestCase action =
        -- is not the case, please email the maintainer.
        E.Handler (\e -> throw (e :: E.AsyncException)),
 
+#if MIN_VERSION_base(4,9,0)
+       E.Handler (\(E.ErrorCallWithLocation msg _) -> return $ Error Nothing msg),
```

Comment:
It's a pity that the location is added as a `String`.  Parsing it could be an option, as we can ensure correctness by having tests.

An other option could still be to remove the special casing of `ErrorCallWithLocation` and adapt the test suite instead.  That way a user would see the source location from the `Show` instance of `ErrorCall`, right?

Are the binaries of the 8.0 RC somewhere btw.  My googling didn't yield anything.